File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
  let answer = 0;
  const obj = {};

  tangerine.forEach((item) => {
    if (obj[item]) obj[item]++;
    else obj[item] = 1;
  });

  const kind = Object.values(obj).sort((a, b) => b - a);

  for (const item of kind) {
    answer++;
    k = k - item;

    if (k <= 0) break;
  }

  return answer;
}

File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    
    obj = {}
    
    for item in tangerine:
        if item not in obj:
            obj[item] = 1
        else:
            obj[item] += 1
    
    value_list = sorted(list(obj.values()), reverse = True)
    
    for item in value_list:
        answer += 1
        k -= item
        
        if k <= 0:
            break
    
    return answer